torch/csrc/jit/codegen/cuda/runtime/random_numbers.cu

```diff
@@ -87,3 +87,50 @@ __device__ float rng_uniform_rangef(
   auto uniform01 = rng_uniformf(rng_result, rng_component);
   return from + range * uniform01;
 }
+
+__device__ float normalf(unsigned int x, unsigned int y) {
+  float u = uniformf(x);
+  float v = uniformf(y) * 6.2831855f;
+
+  return sqrtf(-2.0f * logf(u)) * sinf(v);
```

Comment:
Box muller creates two independent normal random numbers every time. Here, the `sin(v)` and `cos(v)` are two available numbers.

It seems to me the current code structure generates one random number from one function call without caching, so I choose to use the `sin` random number and discard the `cos`  random number.

If we can use both numbers, we may improve the efficiency.

Reply:
Are the two numbers
```
sqrtf(-2.0f * logf(u)) * sinf(v);sqrtf(-2.0f * logf(u)) * sinf(v);
```
and
```
sqrtf(-2.0f * logf(u)) * cosf(v);
```
?

If so, we can pass in a `rng_component` to select which to use.